torch/_dynamo/variables/optimizer.py

```diff
@@ -179,17 +179,6 @@ def mark_static(x):
 
         # Recursively realize the variable trackers for optim.state and
         # optim.param_groups, which recursively install the necessary guards.
-
-        # NB: Its necessary to install the guards for optim.state first.
-        # optim.state is a dict with parameters as keys. Therefore, we just put
-        # ID_MATCH on parameters, instead of TENSOR_MATCH. When we install the
-        # guards for param_groups later, VariableTrackerCache just ensures that
-        # we directly return the cached tensor variable tracker without
-        # inserting the TENSOR_MATCH guard.
-        state_vt = LazyVariableTracker.realize_all(
-            VariableBuilder(tx, AttrSource(self.source, "state"))(self.value.state)
-        )
-
```

Comment:
Can you check that we are still doing DATA_PTR_MATCH guards on the parameters?

Reply:
Yeah we are